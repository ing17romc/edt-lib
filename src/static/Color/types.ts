/**
 * Tipos y contratos para el componente Color.
 */

/**
 * Props para el componente Color.
 * Actualmente no recibe props, pero se deja la interfaz para futuras extensiones.
 */
export type ColorProps = Record<string, never>;
