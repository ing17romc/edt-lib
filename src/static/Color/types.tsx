/**
 * Tipos y contratos para el componente Color.
 */

/**
 * Props para el componente Color.
 */
export interface ColorProps {
  /**
   * Clase CSS personalizada para el contenedor principal
   */
  className?: string;
  
  /**
   * Si es verdadero, muestra solo los colores de fuente
   */
  showFontColors?: boolean;
  
  /**
   * Si es verdadero, muestra solo los colores de fondo
   */
  showBackgroundColors?: boolean;
  
  /**
   * Si es verdadero, muestra las variables de color SCSS
   */
  showScssVariables?: boolean;
}

/**
 * Tipo para las categor√≠as de colores
 */
export type ColorCategory = 'font' | 'background' | 'variable';

/**
 * Interfaz para los grupos de colores
 */
export interface ColorGroup {
  title: string;
  type: ColorCategory;
  items: string[];
}
