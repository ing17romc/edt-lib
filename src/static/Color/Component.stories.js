import React from 'react'

export default {
	title: 'Static/Color'
}

const Template = (args) => {
	const dot = {
		height: '18px',
		width: '18px',
		borderRadius: '50%',
		display: 'inline-block'
	}

	const getP = (style) => <>
		<span className={style}>
			{ style }
		</span>
		{` <span className="${style}"></span>`}
		<hr></hr>
	</>

	const getSpan = (style) => <>
		<span style={dot} className={style}></span>
		{` <span className="${style}"></span>`}
		<hr></hr>
	</>

	const getVariablesColor = (style) => <>
		<span style={dot} className={style}></span>
		{` $${style}` }
		<hr></hr>
	</>

	return <>
		<h1>Font color</h1>
		<hr></hr>
		{getP('font-blue')}
		{getP('font-red')}
		{getP('font-orange')}
		{getP('font-green')}
		{getP('font-white')}
		{getP('font-yellow')}
		{getP('font-gray')}
		<h1>Background color </h1>
		<hr></hr>
		{getSpan('bg-primary-1')}
		{getSpan('bg-primary-2')}
		{getSpan('bg-primary-3')}
		{getSpan('bg-white')}
		{getSpan('bg-light-gray')}
		{getSpan('bg-gray')}
		<h1>SCSS color variables</h1>
		<hr></hr>
		{getVariablesColor('black-color')}
		{getVariablesColor('dark-gray-color')}
		{getVariablesColor('gray-color')}
		{getVariablesColor('light-gray-color')}
		{getVariablesColor('white-color')}
		{getVariablesColor('light-blue-color')}

		{getVariablesColor('dark-blue-color')}
		{getVariablesColor('red-color')}
		{getVariablesColor('orange-color')}
		{getVariablesColor('green-color')}
		{getVariablesColor('yellow-color')}

		{getVariablesColor('primary-color-1')}
		{getVariablesColor('primary-color-2')}
		{getVariablesColor('primary-color-3')}
	</>
}
export const Default = Template.bind({})
