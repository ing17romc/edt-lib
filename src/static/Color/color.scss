@use '../../styles/variables' as *;

.black-color {
    background-color: $black-color !important;
}

.dark-gray-color {
    background-color: $dark-gray-color !important;
}

.gray-color {
    background-color: $gray-color !important;
}

.light-gray-color {
    background-color: $light-gray-color !important;
}

.white-color {
    background-color: $white-color !important;
}

.light-blue-color {
    background-color: $light-blue-color !important;
}

.primary-color-1 {
    background-color: $primary-color !important;
}

.primary-color-2 {
    background-color: $primary-color-2 !important;
}

.primary-color-3 {
    background-color: $primary-color-3 !important;
}

.dark-blue-color {
    background-color: $dark-blue-color !important;
}

.red-color {
    background-color: $red-color !important;
}

.orange-color {
    background-color: $orange-color !important;
}

.green-color {
    background-color: $green-color !important;
}

.yellow-color {
    background-color: $yellow-color !important;
}