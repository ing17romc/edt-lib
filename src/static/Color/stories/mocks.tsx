import type { ColorGroup, ColorProps } from '../types';
import type { Meta } from '@storybook/react';

/**
 * Grupos de colores para mostrar en el componente
 */
export const colorGroups: ColorGroup[] = [
  {
    title: 'Font color',
    type: 'font',
    items: [
      'font-blue',
      'font-red',
      'font-orange',
      'font-green',
      'font-white',
      'font-yellow',
      'font-gray',
    ],
  },
  {
    title: 'Background color',
    type: 'background',
    items: [
      'bg-primary-1',
      'bg-primary-2',
      'bg-primary-3',
      'bg-white',
      'bg-light-gray',
      'bg-gray',
    ],
  },
  {
    title: 'SCSS color variables',
    type: 'variable',
    items: [
      'black-color',
      'dark-gray-color',
      'gray-color',
      'light-gray-color',
      'white-color',
      'light-blue-color',
      'dark-blue-color',
      'red-color',
      'orange-color',
      'green-color',
      'yellow-color',
      'primary-color-1',
      'primary-color-2',
      'primary-color-3',
    ],
  },
];

/**
 * Configuraci칩n por defecto para las historias de Storybook
 */
export const defaultArgs: ColorProps = {
  showFontColors: true,
  showBackgroundColors: true,
  showScssVariables: true,
};

/**
 * Configuraci칩n de controles para Storybook
 */
export const argTypes: Meta<ColorProps>['argTypes'] = {
  className: {
    control: { type: 'text' },
    description: 'Clase CSS personalizada para el contenedor principal',
  },
  showFontColors: {
    control: { type: 'boolean' },
    description: 'Mostrar colores de fuente',
  },
  showBackgroundColors: {
    control: { type: 'boolean' },
    description: 'Mostrar colores de fondo',
  },
  showScssVariables: {
    control: { type: 'boolean' },
    description: 'Mostrar variables SCSS de color',
  },
};

/**
 * Configuraci칩n de par치metros para las historias
 */
export const parameters = {
  layout: 'fullscreen',
  docs: {
    description: {
      component: 'Componente que muestra ejemplos de colores, incluyendo colores de fuente, colores de fondo y variables SCSS de color.',
    },
  },
};
