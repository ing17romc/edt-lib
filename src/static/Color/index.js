import React from 'react'
import './color.scss'

/**
 * @decription TO DO
 *
 * @returns TO DO
 */

const dot = {
	height: '18px',
	width: '18px',
	borderRadius: '50%',
	display: 'inline-block'
}

const getP = (style) => <>
	<span className={style}>
		{ style }
	</span>
	{' <span className="' + style + '"></span>'}
	<hr></hr>
</>

const getSpan = (style) => <>
	<span style={dot} className={style}></span>
	{' <span className="' + style + '"></span>'}
	<hr></hr>
</>

const getVariablesColor = (style) => <>
	<span style={dot} className={style}></span>
	{' $' + style }
	<hr></hr>
</>

const Component = () => <>
	<h1>Font color</h1>
	<hr></hr>
	{getP('font-blue')}
	{getP('font-red')}
	{getP('font-orange')}
	{getP('font-green')}
	{getP('font-white')}
	{getP('font-yellow')}
	{getP('font-gray')}
	<h1>Background color </h1>
	<hr></hr>
	{getSpan('bg-primary-1')}
	{getSpan('bg-primary-2')}
	{getSpan('bg-primary-3')}
	{getSpan('bg-white')}
	{getSpan('bg-light-gray')}
	{getSpan('bg-gray')}
	<h1>SCSS color variables</h1>
	<hr></hr>
	{getVariablesColor('black-color')}
	{getVariablesColor('dark-gray-color')}
	{getVariablesColor('gray-color')}
	{getVariablesColor('light-gray-color')}
	{getVariablesColor('white-color')}
	{getVariablesColor('light-blue-color')}

	{getVariablesColor('dark-blue-color')}
	{getVariablesColor('red-color')}
	{getVariablesColor('orange-color')}
	{getVariablesColor('green-color')}
	{getVariablesColor('yellow-color')}

	{getVariablesColor('primary-color-1')}
	{getVariablesColor('primary-color-2')}
	{getVariablesColor('primary-color-3')}
</>

export default Component
