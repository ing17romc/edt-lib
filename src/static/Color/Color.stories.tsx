import type { Meta, StoryObj } from '@storybook/react';
import './color.scss';
import Color from './index';
import { defaultArgs, argTypes, parameters } from './stories/mocks';

const meta: Meta<typeof Color> = {
  title: 'Static/Color',
  component: Color,
  tags: ['autodocs'],
  args: defaultArgs,
  argTypes,
  parameters,
};

export default meta;

type Story = StoryObj<typeof Color>;

export const Default: Story = {
  args: {
    ...defaultArgs,
  },
};

export const FontColorsOnly: Story = {
  args: {
    showFontColors: true,
    showBackgroundColors: false,
    showScssVariables: false,
  },
  parameters: {
    docs: {
      description: {
        story: 'Muestra solo los colores de fuente.'
      }
    }
  }
};

export const BackgroundColorsOnly: Story = {
  args: {
    showFontColors: false,
    showBackgroundColors: true,
    showScssVariables: false,
  },
  parameters: {
    docs: {
      description: {
        story: 'Muestra solo los colores de fondo.'
      }
    }
  }
};

export const ScssVariablesOnly: Story = {
  args: {
    showFontColors: false,
    showBackgroundColors: false,
    showScssVariables: true,
  },
  parameters: {
    docs: {
      description: {
        story: 'Muestra solo las variables SCSS de color.'
      }
    }
  }
};
