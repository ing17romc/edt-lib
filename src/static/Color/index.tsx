/**
 * Color
 *
 * Este componente muestra ejemplos de colores de fuente, colores de fondo y variables SCSS de color utilizadas en la librería EDT.
 * Útil para visualizar y validar los estilos de color definidos en la librería.
 */
import React from 'react';
import './color.scss';
import type { ColorProps, ColorCategory } from './types';
import { colorGroups } from './stories/mocks';

// Estilos en línea para los puntos de color
const dotStyle: React.CSSProperties = {
  height: '18px',
  width: '18px',
  borderRadius: '50%',
  display: 'inline-block',
  marginRight: '8px',
  verticalAlign: 'middle',
};

/**
 * Renderiza un elemento de color con su clase y código de ejemplo
 */
const renderColorItem = (className: string, type: ColorCategory) => {
  switch (type) {
    case 'font':
      return (
        <div key={className}>
          <span className={className}>{className}</span>
          <code className="code-block">{`<span className="${className}">Texto</span>`}</code>
          <hr className="divider" />
        </div>
      );
    case 'background':
      return (
        <div key={className}>
          <div style={{ display: 'flex', alignItems: 'center', gap: '8px' }}>
            <span style={dotStyle} className={className} />
            <code className="code-block">{`<div className="${className}"></div>`}</code>
          </div>
          <hr className="divider" />
        </div>
      );
    case 'variable':
      return (
        <div key={className}>
          <div style={{ display: 'flex', alignItems: 'center', gap: '8px' }}>
            <span style={{ ...dotStyle, backgroundColor: `var(--${className})` }} />
            <code className="code-block">${className}</code>
          </div>
          <hr className="divider" />
        </div>
      );
    default:
      return null;
  }
};

/**
 * Componente que muestra una paleta de colores con ejemplos de uso
 */
const Color: React.FC<ColorProps> = ({
  className = '',
  showFontColors = true,
  showBackgroundColors = true,
  showScssVariables = true,
}) => {
  // Filtrar los grupos de colores según las props
  const filteredGroups = colorGroups.filter((group) => {
    if (group.type === 'font') return showFontColors;
    if (group.type === 'background') return showBackgroundColors;
    if (group.type === 'variable') return showScssVariables;
    return true;
  });

  return (
    <div 
      className={`color-palette ${className}`}
      data-testid="color-palette"
    >
      {filteredGroups.map((group) => (
        <section key={group.title} className="color-group">
          <h2 className="color-group__title">{group.title}</h2>
          <div className="color-group__items">
            {group.items.map((item) => renderColorItem(item, group.type))}
          </div>
        </section>
      ))}
    </div>
  );
};

export default Color;

