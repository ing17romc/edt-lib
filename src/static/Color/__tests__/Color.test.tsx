import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Color from '../index';
import { mockColorGroups } from './mocks';

describe('Color', () => {
  it('renderiza correctamente con las props por defecto', () => {
    render(<Color />);
    
    // Verificar que se renderizan los títulos de los grupos
    mockColorGroups.forEach(group => {
      expect(screen.getByText(group.title)).toBeInTheDocument();
    });
  });

  it('muestra solo los colores de fuente cuando showFontColors es true', () => {
    render(<Color showFontColors={true} showBackgroundColors={false} showScssVariables={false} />);
    
    // Verificar que solo se muestra el grupo de colores de fuente
    expect(screen.getByText('Font color')).toBeInTheDocument();
    expect(screen.queryByText('Background color')).not.toBeInTheDocument();
    expect(screen.queryByText('SCSS color variables')).not.toBeInTheDocument();
    
    // Verificar que se muestran los colores de fuente
    mockColorGroups
      .find(group => group.type === 'font')
      ?.items.forEach(color => {
        expect(screen.getByText(color)).toBeInTheDocument();
      });
  });

  it('aplica la clase personalizada cuando se proporciona', () => {
    const customClass = 'custom-color-palette';
    render(<Color className={customClass} />);
    
    const container = screen.getByTestId('color-palette');
    expect(container).toHaveClass(customClass);
  });

  it('no renderiza los grupos cuando todas las props son falsas', () => {
    render(<Color 
      showFontColors={false} 
      showBackgroundColors={false} 
      showScssVariables={false} 
    />);
    
    // Verificar que no se renderiza ningún grupo
    mockColorGroups.forEach(group => {
      expect(screen.queryByText(group.title)).not.toBeInTheDocument();
    });
  });

  it('coincide con el snapshot', () => {
    const { container } = render(<Color />);
    expect(container).toMatchSnapshot();
  });
});
