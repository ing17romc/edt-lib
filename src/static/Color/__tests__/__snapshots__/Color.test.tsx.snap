// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color coincide con el snapshot 1`] = `
<div>
  <div
    class="color-palette "
    data-testid="color-palette"
  >
    <section
      class="color-group"
    >
      <h2
        class="color-group__title"
      >
        Font color
      </h2>
      <div
        class="color-group__items"
      >
        <div>
          <span
            class="font-blue"
          >
            font-blue
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-blue"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-red"
          >
            font-red
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-red"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-orange"
          >
            font-orange
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-orange"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-green"
          >
            font-green
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-green"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-white"
          >
            font-white
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-white"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-yellow"
          >
            font-yellow
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-yellow"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
        <div>
          <span
            class="font-gray"
          >
            font-gray
          </span>
          <code
            class="code-block"
          >
            &lt;span className="font-gray"&gt;Texto&lt;/span&gt;
          </code>
          <hr
            class="divider"
          />
        </div>
      </div>
    </section>
    <section
      class="color-group"
    >
      <h2
        class="color-group__title"
      >
        Background color
      </h2>
      <div
        class="color-group__items"
      >
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-primary-1"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-primary-1"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-primary-2"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-primary-2"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-primary-3"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-primary-3"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-white"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-white"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-light-gray"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-light-gray"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              class="bg-gray"
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle;"
            />
            <code
              class="code-block"
            >
              &lt;div className="bg-gray"&gt;&lt;/div&gt;
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
      </div>
    </section>
    <section
      class="color-group"
    >
      <h2
        class="color-group__title"
      >
        SCSS color variables
      </h2>
      <div
        class="color-group__items"
      >
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--black-color);"
            />
            <code
              class="code-block"
            >
              $
              black-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--dark-gray-color);"
            />
            <code
              class="code-block"
            >
              $
              dark-gray-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--gray-color);"
            />
            <code
              class="code-block"
            >
              $
              gray-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--light-gray-color);"
            />
            <code
              class="code-block"
            >
              $
              light-gray-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--white-color);"
            />
            <code
              class="code-block"
            >
              $
              white-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--light-blue-color);"
            />
            <code
              class="code-block"
            >
              $
              light-blue-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--dark-blue-color);"
            />
            <code
              class="code-block"
            >
              $
              dark-blue-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--red-color);"
            />
            <code
              class="code-block"
            >
              $
              red-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--orange-color);"
            />
            <code
              class="code-block"
            >
              $
              orange-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--green-color);"
            />
            <code
              class="code-block"
            >
              $
              green-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--yellow-color);"
            />
            <code
              class="code-block"
            >
              $
              yellow-color
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--primary-color-1);"
            />
            <code
              class="code-block"
            >
              $
              primary-color-1
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--primary-color-2);"
            />
            <code
              class="code-block"
            >
              $
              primary-color-2
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
        <div>
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <span
              style="height: 18px; width: 18px; border-radius: 50%; display: inline-block; margin-right: 8px; vertical-align: middle; background-color: var(--primary-color-3);"
            />
            <code
              class="code-block"
            >
              $
              primary-color-3
            </code>
          </div>
          <hr
            class="divider"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
